lib/order.rb

```diff
@@ -0,0 +1,36 @@
+class Order
+  attr_reader :items, :total
+
+  def initialize(menu = Menu.new)
+    @items = []
+    @total = 0
+    @menu = menu
+    @menu_items = menu.dishes
+  end
+
+  def add_item(dish, quantity = 1)
+    # This method is currently doing too much and needs refactoring
+    item = @menu_items.select { |menu_item| menu_item[:name] == dish }
+    item = item.first
+    item[:quantity] = quantity
```

Comment:
Can cause issue if you add the same item twice, the previous quantity will get overwritten

Reply:
Yes, if I had more time I would have added some kind of conditional to check if this dish is already inside the order's `@items` array and if so, only updating quantity instead of adding a new item.